### PR TITLE
CPM-751: Add generator list page

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**

Screen when there are no generator created yet
Screenshot from 2022-10-27 10-49-01

Screen when there is one generator
Screenshot from 2022-10-27 10-46-13

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 1]
```diff
@@ -1,39 +1,83 @@
-import React from 'react';
-import {Common} from '../components';
-import {PageHeader, PimView, useTranslate} from '@akeneo-pim-community/shared';
-import {Breadcrumb, Button} from 'akeneo-design-system';
+import React, {useCallback, useMemo} from 'react';
+import {useTranslate, useUserContext} from '@akeneo-pim-community/shared';
+import {AttributesIllustration, Button, Helper, Table} from 'akeneo-design-system';
+import {useGetGenerators} from '../hooks/useGetGenerators';
```
Comment (antoinearrive) [inline components/identifier-generator/front/src/feature/pages/ListPage.tsx:4]: could the import be improved by importing it into index.ts?

export * from './useGetGenerators';

**Review by antoinearrive — APPROVED** [review]

File: components/identifier-generator/front/src/feature/components/ListSkeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import {Styled} from './Styled';
+
+const ListSkeleton: React.FC = () => (
+  <>
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/components/ListSkeleton.tsx:5]: <> useful ?

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 3]
```diff
@@ -1,39 +1,83 @@
...
-        </PageHeader.UserActions>
-        <PageHeader.Actions>
-          <Button onClick={onCreate} disabled={!isCreateEnabled}>
-            {translate('pim_common.create')}
-          </Button>
-        </PageHeader.Actions>
-        <PageHeader.Title>{translate('pim_title.akeneo_identifier_generator_index')}</PageHeader.Title>
-      </PageHeader>
+      <Header>
+        <Button onClick={onCreate} disabled={!isGeneratorListEmpty}>
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/ListPage.tsx:27]: The condition is more "generators.length >= 1" (for the future)

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 4]
```diff
@@ -1,39 +1,83 @@
...
+      <Styled.Container>
+        <Table>
+          <Table.Header>
+            <Table.HeaderCell>{translate('pim_common.label')}</Table.HeaderCell>
+            <Table.HeaderCell>{translate('pim_identifier_generator.list.identifier')}</Table.HeaderCell>
+            <Table.HeaderCell></Table.HeaderCell>
+          </Table.Header>
+          <Table.Body>
+            {isGeneratorListEmpty && !isLoading && (
+              <Styled.NoIdentifierMessage>
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/ListPage.tsx:40]: There is a DSM component for that no ?

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 5]
```diff
@@ -1,39 +1,83 @@
...
+                <AttributesIllustration />
+                <Styled.Title>{translate('pim_identifier_generator.list.first_generator')}</Styled.Title>
+                <Styled.HelpCenterLink
+                  href="https://help.akeneo.com/pim/serenity/articles/understand-data-quality.html"
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/ListPage.tsx:44]: not sure

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 6]
```diff
@@ -1,39 +1,83 @@
...
+            )}
+            {isLoading && <ListSkeleton />}
+            {!isGeneratorListEmpty && (
+              <>
+                <tr>
+                  <td colSpan={3}>
+                    <Helper level="info">
+                      {translate('pim_identifier_generator.list.create_info')}
+                      <a
+                        href="https://help.akeneo.com/pim/serenity/articles/understand-data-quality.html"
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/ListPage.tsx:59]: idem

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 7]
```diff
@@ -1,39 +1,83 @@
...
+                <Styled.HelpCenterLink
+                  href="https://help.akeneo.com/pim/serenity/articles/understand-data-quality.html"
+                  target="_blank"
+                >
+                  {translate('pim_identifier_generator.list.check_help_center')}
+                </Styled.HelpCenterLink>
+              </Styled.NoIdentifierMessage>
+            )}
+            {isLoading && <ListSkeleton />}
+            {!isGeneratorListEmpty && (
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/ListPage.tsx:52]: Weird to have this helper when you already created once

File: components/identifier-generator/front/src/feature/pages/styles/ListPageStyled.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import styled from 'styled-components';
+import {getColor} from 'akeneo-design-system';
+
+const NoIdentifierMessage = styled.div`
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/styles/ListPageStyled.ts:4]: https://dsm.akeneo.com/?path=/docs/components-placeholder--standard

**Review by pierallard — APPROVED** [review]

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx (hunk 7 above)
Comment (cemorin) [reply components/identifier-generator/front/src/feature/pages/ListPage.tsx:52]: It's in the mockups, to inform the user that they can only create one generator for now.

Screenshot from 2022-11-02 09-33-39

**Pull request merged**